shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -1068,30 +1087,32 @@ object Types {
     def paramNum: Int
 
   case class TypeLambda(paramNames: List[TypeName])(
-    @constructorOnly paramTypeBoundsExp: TypeLambda => List[TypeBounds],
-    @constructorOnly resultTypeExp: TypeLambda => Type
+    paramTypeBoundsExp: TypeLambda => List[TypeBounds],
```

Comment:
curious as to why this had to change when you have implemented lookahead in the treeunpickler

Reply:
It's not so much for the `TreeUnpickler` changes as for the `TypeLambdaTree.toType` changes. Since we must be able to execute `toType` without looking at other files.

Reply:
The external references can be read lazily. The problem is with `integrate`, which is too eager in trying to resolve aliases and what not.

To avoid this, we should have a "syntactic" version of type substitution that only looks at local symbols, and leaves everything else alone. But that's another story.